src/Provider/AbstractProvider.php

```diff
@@ -312,35 +341,53 @@ protected function getAccessTokenMethod()
         return 'POST';
     }
 
-    public function getAccessToken($grant = 'authorization_code', array $params = [])
+    /**
+     * Builds the access token URL's query string.
+     *
+     * @param array $params Query parameters
+     * @return string Query string
+     */
+    protected function getAccessTokenQuery(array $params)
+    {
+        return http_build_query($params);
+    }
+
+    /**
+     * Checks that a provided grant is valid, or attempts to produce one if the
+     * provided grant is a string.
+     *
+     * @param mixed $grant
+     * @return GrantInterface
+     */
+    protected function verifyGrant($grant)
     {
         if (is_string($grant)) {
-            $grant = $this->grantFactory->getGrant($grant);
-        } else {
-            $this->grantFactory->checkGrant($grant);
+            return $this->grantFactory->getGrant($grant);
```

Comment:
@rtheunissen but if the grant is a string, it gets turned into an instance here.